### PR TITLE
0.1x spec

### PR DESCRIPTION
**Pull request by technoweenie**
Runs rspec on 0.1x, to ensure the `error_spec.rb` file introduced in #1054 is actually run.

Also gitignores the `./docs` dir. It contains the github pages files, which didn't exist back then.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,6 +27,8 @@ group :test do
   gem 'patron', '>= 0.4.2', :platforms => :ruby
   gem 'rack-test', '>= 0.6', :require => 'rack/test'
   gem 'rest-client', '~> 1.6.0', :platforms => [:jruby, :ruby_18]
+  gem 'rspec', '~> 3.7'
+  gem 'rspec_junit_formatter', '~> 0.4'
   gem 'simplecov'
```
Comment (olleolleolle) [inline Gemfile]: Will you begin using the JUnit XML output for anything, now outside of Circle CI?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  # the `--only-failures` and `--next-failure` CLI options. We recommend
+  # you configure your source control system to ignore this file.
+  config.example_status_persistence_file_path = "spec/examples.txt"
+
+  # Limits the available syntax to the non-monkey patched syntax that is
+  # recommended. For more details, see:
+  #   - http://rspec.info/blog/2012/06/rspecs-new-expectation-syntax/
+  #   - http://www.teaisaweso.me/blog/2013/05/27/rspecs-new-message-expectation-syntax/
+  #   - http://rspec.info/blog/2014/05/notable-changes-in-rspec-3/#zero-monkey-patching-mode
+  config.disable_monkey_patching!
```
Comment (olleolleolle) [inline spec/spec_helper.rb]: Enable this one, it's good.

Comment (technoweenie) [reply spec/spec_helper.rb]: Sure, but it's not enabled on `master` right now :)

File: Gemfile (hunk 1 above)
Comment (technoweenie) [reply Gemfile]: Oh that was just a circle ci thing? I'll remove it from this pr.

**Pull request merged**